console/src/main/java/org/matrix/console/store/LoginStorage.java

```diff
@@ -189,58 +143,37 @@ public boolean addCredentials(Credentials credentials) {
-
-        String usernames = prefs.getString(PREFS_KEY_USERNAMES, null);
-        String servers = prefs.getString(PREFS_KEY_HOME_SERVERS, null);
-        String tokens = prefs.getString(PREFS_KEY_ACCESS_TOKENS, null);
-
-        // has some credentials
-        if ((null == usernames) || (null == servers) || (null == tokens)) {
-            return false;
-        }
-
-        ArrayList<String> usernamesList = deserialize(usernames);
-        int pos = usernamesList.indexOf(credentials.userId);
-
-        if (pos >= 0) {
-            SharedPreferences.Editor e = prefs.edit();
-
-            usernamesList.remove(pos);
-            if (0 == usernamesList.size()) {
-                e.putString(PREFS_KEY_USERNAMES, null);
-            } else {
-                e.putString(PREFS_KEY_USERNAMES, serialize(usernamesList));
-            }
-
-            ArrayList<String> homeServersList = deserialize(servers);
-            homeServersList.remove(pos);
-            if (0 == homeServersList.size()) {
-                e.putString(PREFS_KEY_HOME_SERVERS, null);
-            } else {
-                e.putString(PREFS_KEY_HOME_SERVERS, serialize(homeServersList));
+    public Boolean removeCredentials(HomeserverConnectionConfig config) {
+        if (null != config) {
+            SharedPreferences prefs = mContext.getSharedPreferences(PREFS_LOGIN, Context.MODE_PRIVATE);
+            SharedPreferences.Editor editor = prefs.edit();
+
+            ArrayList<HomeserverConnectionConfig> configs = getCredentialsList();
+            ArrayList<JSONObject> serialized = new ArrayList<JSONObject>(configs.size());
+
+            boolean found = false;
+            try {
+                for (HomeserverConnectionConfig c : configs) {
+                    if (c.getCredentials().userId.equals(config.getCredentials().userId)) {
+                        found = true;
+                    } else {
+                        serialized.add(c.toJson());
+                    }
+                }
+            } catch (JSONException e) {
+                throw new RuntimeException("Failed to serialize connection config");
             }
 
-            ArrayList<String> tokensList = deserialize(tokens);
-            tokensList.remove(pos);
-            if (0 == tokensList.size()) {
-                e.putString(PREFS_KEY_ACCESS_TOKENS, null);
-            } else {
-                e.putString(PREFS_KEY_ACCESS_TOKENS, serialize(tokensList));
-            }
+            if (!found) return false;
 
-            // remove the old storage
-            e.putString(PREFS_KEY_ACCESS_TOKEN, null);
-            e.putString(PREFS_KEY_HOME_SERVER, null);
-            e.putString(PREFS_KEY_USERNAME, null);
+            editor.putString(PREFS_KEY_CONNECTION_CONFIGS, new JSONArray(serialized).toString());
 
-            return e.commit();
+            return editor.commit();
         }
 
-        return true;
+        return false;
```

Comment:
Here you return `false` but in the prev diff it's `true`? What does this `boolean` even mean?